main/vstring.h

```diff
@@ -30,7 +30,7 @@
 #endif
 #ifdef VSTRING_PUTC_MACRO
 #define vStringPut(s,c) \
-	(void)(((s)->length + 1 == (s)->size ? vStringResize ((s), (s)->size * 2) : 0), \
+	(void)(((s)->length + 1 == (s)->size ? vStringResize ((s), (s)->size * 2) : (void) 0 ), \
 	((s)->buffer [(s)->length] = (c)), \
 	((c) == '\0' ? 0 : ((s)->buffer [++(s)->length] = '\0')))
 #endif
```

Comment:
Instead how about following code?
```
diff --git a/main/vstring.h b/main/vstring.h
index 203a5c2..2300478 100644
--- a/main/vstring.h
+++ b/main/vstring.h
@@ -29,10 +29,14 @@
 # define VSTRING_PUTC_MACRO 1
 #endif
 #ifdef VSTRING_PUTC_MACRO
-#define vStringPut(s,c) \
-	(void)(((s)->length + 1 == (s)->size ? vStringResize ((s), (s)->size * 2) : 0), \
-	((s)->buffer [(s)->length] = (c)), \
-	((c) == '\0' ? 0 : ((s)->buffer [++(s)->length] = '\0')))
+#define vStringPut(s,c)							\
+	do {										\
+		if ((s)->length + 1 == (s)->size)		\
+			vStringResize ((s), (s)->size * 2);	\
+		(s)->buffer [(s)->length] = (c);		\
+		if ((c) != '\0')						\
+			(s)->buffer [++(s)->length] = '\0';	\
+	} while (0);
 #endif
 
 #define vStringValue(vs)      ((vs)->buffer)
```

I believe C compilers may generate the same code.
How do you think?